src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -128,244 +129,225 @@ export class TerminalLaunchHelpAction extends Action {
+	options: IAction2Options & { run: (c: ITerminalServicesCollection, accessor: ServicesAccessor, args?: unknown) => void | Promise<unknown> }
+): IDisposable {
+	// Set defaults
+	options.f1 = options.f1 ?? true;
+	options.category = options.category ?? category;
+	options.precondition = options.precondition ?? TerminalContextKeys.processSupported;
+	// Remove run function from options so it's not passed through to registerAction2
+	const runFunc = options.run;
+	const strictOptions: IAction2Options & { run?: (c: ITerminalServicesCollection, accessor: ServicesAccessor, args?: unknown) => void | Promise<unknown> } = options;
+	delete (strictOptions as IAction2Options & { run?: (c: ITerminalServicesCollection, accessor: ServicesAccessor, args?: unknown) => void | Promise<unknown> })['run'];
+	// Register
+	return registerAction2(class extends Action2 {
+		constructor() {
+			super(strictOptions as IAction2Options);
+		}
+		run(accessor: ServicesAccessor, args?: unknown) {
+			return runFunc(getTerminalServices(accessor), accessor, args);
+		}
+	});
+}
+
+function registerActiveInstanceAction(
+	options: IAction2Options & { run: (activeInstance: ITerminalInstance, c: ITerminalServicesCollection, accessor: ServicesAccessor, args?: unknown) => void | Promise<unknown> }
+): IDisposable {
+	const originalRun = options.run;
+	return registerTerminalAction({
+		...options,
+		run: (c, accessor, args) => {
+			const activeInstance = c.service.activeInstance;
+			if (activeInstance) {
+				return originalRun(activeInstance, c, accessor, args);
+			}
 		}
-		async run(accessor: ServicesAccessor) {
-			const terminalService = accessor.get(ITerminalService);
-			const terminalGroupService = accessor.get(ITerminalGroupService);
-			if (terminalService.isProcessSupportRegistered) {
-				const instance = await terminalService.createTerminal({ location: terminalService.defaultLocation });
+	});
+}
+
+interface ITerminalServicesCollection {
+	service: ITerminalService;
+	groupService: ITerminalGroupService;
+	instanceService: ITerminalInstanceService;
+	editorService: ITerminalEditorService;
+	profileService: ITerminalProfileService;
+	profileResolverService: ITerminalProfileResolverService;
+}
+
+function getTerminalServices(accessor: ServicesAccessor): ITerminalServicesCollection {
+	return {
+		service: accessor.get(ITerminalService),
+		groupService: accessor.get(ITerminalGroupService),
+		instanceService: accessor.get(ITerminalInstanceService),
+		editorService: accessor.get(ITerminalEditorService),
+		profileService: accessor.get(ITerminalProfileService),
+		profileResolverService: accessor.get(ITerminalProfileResolverService)
+	};
+}
+
+export function registerTerminalActions() {
+	registerTerminalAction({
+		id: TerminalCommandId.NewInActiveWorkspace,
+		title: { value: localize('workbench.action.terminal.newInActiveWorkspace', "Create New Terminal (In Active Workspace)"), original: 'Create New Terminal (In Active Workspace)' },
+		run: async (c) => {
```

Comment:
I like how concise `c` is here, but it makes it a little confusing. Could it instead be called `collection` or `services`?

Reply:
`services.service` didn't feel nice, part of this was to simplify the run implementations as well. I think `c` is alright considering it's used as a standard consistently throughout the file